server/controllers/contest.js

```diff
@@ -104,3 +107,74 @@ exports.updateContest = asyncHandler(async (req, res, next) => {
     throw new Error("Could not update contest at this time, Please try again");
   }
 });
+
+exports.addCustomer = asyncHandler(async (req, res, next) => {
+  const { source, price, save } = req.body;
+
+  const customer = await Customer.findOne({
+    user_id: req.user.id,
+  });
+
+  const user = await User.findById(req.user.id);
+
+  if (customer) {
+    if (price > 0) {
+      const paymentCreate = await stripe.charges.create({
+        amount: 500,
+        currency: "usd",
+        // because we are in a test env source:"tok_visa" must be sent instead of customer_id
+        source: "tok_visa",
+      });
+
+      res.status(201).json({
+        success: {
+          payment: paymentCreate,
+          message: "Payment was successful",
+        },
+      });
+    } else {
+      res.status(400);
+      throw new Error("Credit Card already exists.");
+    }
+  }
+
+  const newCustomer = await stripe.customers.create({
```

Comment:
Because there's no return logic in the above block, this new customer will always be created.